python/pyiceberg/expressions/__init__.py

```diff
@@ -18,18 +18,34 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
-from dataclasses import dataclass
 from functools import reduce
-from typing import ClassVar, Generic, TypeVar
-
-from pyiceberg.expressions.literals import Literal
+from typing import (
+    Any,
+    Generic,
+    Iterable,
+    Set,
+    Type,
+    Union,
+)
+
+from pyiceberg.expressions.literals import Literal, literal
 from pyiceberg.files import StructProtocol
 from pyiceberg.schema import Accessor, Schema
+from pyiceberg.typedef import T
```

Comment:
If this is going to be shared, then maybe we should name it something other than the very generic `T`. `L` seems like one that wouldn't be used often.